ios/RNFIRMessaging.m

```diff
@@ -169,11 +174,45 @@ + (void)willPresentNotification:(UNNotification *)notification withCompletionHan
   [[NSNotificationCenter defaultCenter] postNotificationName:FCMNotificationReceived object:self userInfo:@{@"data": data, @"completionHandler": completionHandler}];
 }
 
++ (void)didRegisterUserNotificationSettings:(__unused UIUserNotificationSettings *)notificationSettings
+{
+  if ([UIApplication instancesRespondToSelector:@selector(registerForRemoteNotifications)]) {
+    [RCTSharedApplication() registerForRemoteNotifications];
+    [[NSNotificationCenter defaultCenter] postNotificationName:RNFIRRegisterUserNotificationSettings
```

Comment:
why post another notification and handle it in the same file?
Can this function directly resolve/reject the promise?